draft-kinnear-webtransport-http2.md

```diff
@@ -63,459 +78,248 @@ Discussion of this draft takes place on the WebTransport mailing list
+client, who sends an extended CONNECT request {{!RFC8441}}.  If the server
+accepts the request, an WebTransport session is established.  The resulting
+stream will be further referred to as a *CONNECT stream*, and its stream ID is
+used to uniquely identify a given WebTransport session within the connection.
+The ID of the CONNECT stream that established a given WebTransport session will
+be further referred to as a *Session ID*.
 
-After negotiating the use of this extension, clients initiate one or more
-WebTransport Connect Streams to a Http2Transport Server. Http2Transport servers
-are identified by a pair of authority value and path value (defined in
-{{!RFC3986}} Sections 3.2 and 3.3 respectively). The client uses the extended
-CONNECT method with a :protocol token "webtransport" to establish a WebTransport
-Connect Stream. This stream is only used to establish a WebTransport session and
-is not intended for data exchange.
+After the session is established, the peers can exchange data using the
+following mechanisms:
 
+* Both client and server can create a bidirectional or unidirectional stream
+  using a new HTTP/2 extension frame (WT_STREAM)
+* A datagram can be sent using a a new HTTP/2 extension frame WT_DATAGRAM.
 
-## WebTransport Streams
+An WebTransport session is terminated when the CONNECT stream that created it
+is closed.
 
-Following the establishment of a WebTransport Connect stream, either the client
-or the server can initiate a WebTransport Stream by sending the WTHEADERS frame,
-defined in {{the-wtheaders-frame}}. This frame references an open WebTransport
-Connect stream which is used by any intermediaries to correctly forward the
-stream to the destination endpoint. The only frames allowed on WebTransport
-Streams are WTHEADERS, CONTINUATION, DATA and any negotiated extension frames.
+# Session Establishment
 
+## Establishing a Transport-Capable HTTP/2 Connection
 
-## Negotiation
+In order to indicate support for WebTransport, both the client and the server
+MUST send a SETTINGS_ENABLE_WEBTRANSPORT value set to "1" in their SETTINGS
+frame.  Endpoints MUST NOT use any WebTransport-related functionality unless
+the parameter has been negotiated.
 
-Clients negotiate the use of WebTransport over HTTP/2 using both the SETTINGS
-frame and one or more extended CONNECT requests as defined in {{!RFC8441}}.
+## Extended CONNECT in HTTP/2
 
-Use of the extended CONNECT method extension requires the
-SETTINGS_ENABLE_CONNECT_PROTOCOL parameter to be received by a client prior to
-its use. An endpoint that supports receiving the extended CONNECT method SHOULD
-send this setting with a value of 1.
+{{!RFC8441}} defines an extended CONNECT method in Section 4, enabled by the
+SETTINGS_ENABLE_CONNECT_PROTOCOL parameter.  An endpoint doesn not need to send
+both SETTINGS_ENABLE_CONNECT_PROTOCOL and SETTINGS_ENABLE_WEBTRANSPORT; the
+SETTINGS_ENABLE_WEBTRANSPORT setting implies that an endpoint supports extended
+CONNECT.
```

Comment:
That's kind of an interesting fallout from this, but seems fine

Reply:
Yeah, the h3 draft does this, but maybe makes more sense there because the extended CONNECT isn't defined in any specification for h3 yet.